dbgd.c

```diff
@@ -298,3 +300,65 @@ static int dbgd_show_front_screen(Dbg__Request *req, Dbg__Response *res)
 
     return DBG__RESPONSE__TYPE__OK;
 }
+
+static int dbgd_call(Dbg__Request *req, Dbg__Response *res)
+{
+    if (!req->has_address)
+        return DBG__RESPONSE__TYPE__ERROR_INCOMPLETE_REQUEST;
+
+    // These variables will be used as parameters for inline assembly.
+    // We make them static as the stack pointer will be changed.
+    // So we want to address them globally.
+    static uint32_t stack_pointer;
+    static uint32_t stack_backup;
+    static uint32_t address;
+
+    // Allocate a stack for working and space for supplied data
+    size_t stack_size = 0x1000;
+    if (req->has_data) {
+      stack_size += req->data.len;
+    }
+    uint8_t* stack_data = malloc(stack_size);
+
+    // Push optional stack contents, starting at top of stack
+    stack_pointer = (uint32_t)&stack_data[stack_size];
+    if (req->has_data) {
+      stack_pointer -= req->data.len;
+      memcpy((void*)stack_pointer, req->data.data, req->data.len);
+    }
+
+    // Set address to call
+    address = req->address;
+
+    asm("pusha\n"
```

Comment:
Formatting nit: Please remove blank lines and align comments on the RHS

Reply:
Will do.